astropy/coordinates/transformations/function.py

```diff
@@ -56,16 +54,6 @@ class FunctionTransform(CoordinateTransform):
     def __init__(self, func, fromsys, tosys, priority=1, register_graph=None):
         if not callable(func):
             raise TypeError("func must be callable")
-
-        with suppress(TypeError):
-            sig = signature(func)
-            kinds = [x.kind for x in sig.parameters.values()]
-            if (
-                len(x for x in kinds if x == sig.POSITIONAL_ONLY) != 2
```

Comment:
Trying to get the length of a generator is guaranteed to raise a `TypeError`, which is then caught by the context manager. The `TypeError` could be avoided by using a list comprehension instead of a generator expression, but then trying to access `sig.POSITIONAL_ONLY` would trigger an `AttributeError` instead. This code block is therefore completely broken.